src/v/kafka/server/usage_manager.cc

```diff
@@ -253,15 +256,59 @@ ss::future<> usage_manager::accounting_fiber::stop() {
     }
 }
 
-ss::future<> usage_manager::accounting_fiber::close_window() {
+void usage_manager::accounting_fiber::close_window() {
     const auto now = ss::lowres_system_clock::now();
-    _buckets[_current_window].u = co_await _um.map_reduce0(
-      [](usage_manager& um) { return um.sample(); },
-      _buckets[_current_window].u,
-      [](const usage& acc, const usage& x) { return acc + x; });
-    _buckets[_current_window].end = epoch_time_secs(now);
+    const auto now_ts = epoch_time_secs(now);
+    const auto before_close_idx = _current_window;
+    _buckets[before_close_idx].end = now_ts;
     _current_window = (_current_window + 1) % _buckets.size();
     _buckets[_current_window].reset(now);
+    /// The timer must progress so subsequent windows may be closed, async work
+    /// may hold that up for a significant amount of time, therefore async work
+    /// will be dispatched in the background and the result set written to the
+    /// correct bucket when it is eventually computed
+    ssx::spawn_with_gate(_gate, [this, before_close_idx, now_ts] {
```

Comment:
what if broker goes down and new leader comes up after the current window close? Will we still get the measurement?

Reply:
So you may query any node for cloud metrics they should all report the same values (not accounting for time skew across buckets)